### PR TITLE
Generatedprocessordocs rearrange sidebar fn

### PR DESCRIPTION
**Pull request by nmcspadden**
The purpose of this is to separate out some of the existing functionality into smaller functions, for added testability in the future. The primary and more immediate objective is to allow taking different actions based on the results. The next PR in this stack will be adding an argument to select a specific processor to update.

Testing by running:
```
./generate_processor_docs.py 1.0.5
Cloning AutoPkg wiki..

Cloned to /var/folders/th/3tqg05g54f73x0k_mstm5wy1_0g_xw/T/tmpW_spSy.


commit 11049654a1f70e98b704421d81fd55f933053e25
Author: Nick McSpadden <nmcspadden@gmail.com>
Date:   Sun May 12 20:58:52 2019 -0700

    Updating Wiki docs for release 1.0.5

diff --git a/Processor-AppPkgCreator.md b/Processor-AppPkgCreator.md
index f2a1a4f..09abb19 100644
--- a/Processor-AppPkgCreator.md
+++ b/Processor-AppPkgCreator.md
@@ -12,7 +12,7 @@ Calls autopkgserver to create a package from an application.
     - **description:** Path to an application to be packaged. Can be on a disk image and globbed. If not set, defaults to %pathname%/\*.app. Typically %pathname% points to a disk image downloaded in a prior recipe step.
 - **version:**
     - **required:** False
-    - **description:** Version of the app. If not set, will be extracted from theCFBundleShortVersionString in the app's Info.plist.
+    - **description:** Version of the app. If not set, will be extracted from the CFBundleShortVersionString in the app's Info.plist.
...
...
diff --git a/_Sidebar.md b/_Sidebar.md
index a423a47..75b75c6 100644
--- a/_Sidebar.md
+++ b/_Sidebar.md
@@ -29,10 +29,10 @@
       * [[AppDmgVersioner|Processor-AppDmgVersioner]]
       * [[AppPkgCreator|Processor-AppPkgCreator]]
       * [[BrewCaskInfoProvider|Processor-BrewCaskInfoProvider]]
-      * [[CodeSignatureVerifier|Processor-CodeSignatureVerifier]]
-      * [[Copier|Processor-Copier]]
       * [[CURLDownloader|Processor-CURLDownloader]]
       * [[CURLTextSearcher|Processor-CURLTextSearcher]]
+      * [[CodeSignatureVerifier|Processor-CodeSignatureVerifier]]
+      * [[Copier|Processor-Copier]]
       * [[DmgCreator|Processor-DmgCreator]]
       * [[DmgMounter|Processor-DmgMounter]]
       * [[EndOfCheckPhase|Processor-EndOfCheckPhase]]
@@ -42,8 +42,8 @@
       * [[FlatPkgPacker|Processor-FlatPkgPacker]]
       * [[FlatPkgUnpacker|Processor-FlatPkgUnpacker]]
       * [[GitHubReleasesInfoProvider|Processor-GitHubReleasesInfoProvider]]
-      * [[Installer|Processor-Installer]]
       * [[InstallFromDMG|Processor-InstallFromDMG]]
+      * [[Installer|Processor-Installer]]
       * [[MunkiCatalogBuilder|Processor-MunkiCatalogBuilder]]
       * [[MunkiImporter|Processor-MunkiImporter]]
       * [[MunkiInfoCreator|Processor-MunkiInfoCreator]]
@@ -63,9 +63,9 @@
       * [[SparkleUpdateInfoProvider|Processor-SparkleUpdateInfoProvider]]
       * [[StopProcessingIf|Processor-StopProcessingIf]]
       * [[Symlinker|Processor-Symlinker]]
-      * [[Unarchiver|Processor-Unarchiver]]
       * [[URLDownloader|Processor-URLDownloader]]
       * [[URLTextSearcher|Processor-URLTextSearcher]]
+      * [[Unarchiver|Processor-Unarchiver]]
       * [[Versioner|Processor-Versioner]]
   * **Development**
       * [[Packaging releases on GitHub|Packaging-AutoPkg-For-Release-on-GitHub]]

-------------------------------------------------------------------

Shown above is the commit log for the changes to the wiki markdown.
Type 'push' to accept and push the changes to GitHub. The wiki repo
local clone can be also inspected at:
/var/folders/th/3tqg05g54f73x0k_mstm5wy1_0g_xw/T/tmpW_spSy.
```

### CODE REVIEW

**Comment by nmcspadden** [comment]
Some day I'll figure out how to correctly squash all this without creating a mess of commits.

**Comment by nmcspadden** [comment]
Ping?

File: Scripts/generate_processor_docs.py [hunk 1]
```diff
@@ -97,6 +97,43 @@ def indent_length(line_str):
     return len(line_str) - len(line_str.lstrip())
 
 
+def generate_sidebar(sidebar_path):
+    """Generate new _Sidebar.md contents."""
+    # Generate the Processors section of the Sidebar
+    processor_heading = "  * **Processor Reference**"
+    toc_string = ""
+    toc_string += processor_heading + "\n"
+    for processor_name in sorted(processor_names()):
```
Comment (gregneagle) [inline Scripts/generate_processor_docs.py]: See https://github.com/autopkg/autopkg/commit/7638c1900f8b5f264e565511ad025ba2f57dca68

**Comment by nmcspadden** [comment]
Yeesh, I've lost all my git skills.

**Comment by gregneagle** [comment]
This tool is definitely not a high priority to work on...

**Comment by nmcspadden** [comment]
I already have the work done, it's just a matter of submitting the PRs correctly so that git doesn't bring along every rebase.

**Pull request closed without merging**